updated "use server" docs with data api usage, single flight actions, and minor recommendations (#723)
solidjs/solid-docs-next: src/routes/solid-start/reference/server/use-server.mdx

logHello("Hello");
```

In both of these examples, the `logHello` function, it would only show in the server console regardless of whether rendering was on the server or in the browser.

## Usage with Data APIs

> Server functions can be used for fetching data and performing actions on the server. The following examples show how to use server functions alongside solid-router's data APIs.

Reviewer: ```suggestion
Server functions can be used for fetching data and performing actions on the server. 
The following examples show how to use server functions alongside solid-router's data APIs.
```

```tsx {3}
const getUser = cache((id) => {
  "use server";
  return db.getUser(id);
}, "users");

const updateUser = action(async (id, data) => {
  "use server"
  await db.setUser(id, data);
  throw redirect("/", { revalidate: getUser.keyFor(id) });
});

```

> When `getUser` or `updateUser` are invoked on the client, an http request will be made to the server, which invokes the corresponding server function.

Reviewer: ```suggestion
When `getUser` or `updateUser` are invoked on the client, an http request will be made to the server, which calls the corresponding server function.
```

## Single-flight actions

> In the above example, when the `updateUser` action is called, a redirect is thrown on the server. Start can handle this redirect on the server instead of propagating it to the client. The data for the redirected page is fetched and streamed to the client in the same http request as the `updateUser` action, instead of the client having to make a separate http request for the redirected page.

Reviewer: ```suggestion
In the above example, when the `updateUser` action is called, a redirect is thrown on the server. 
Solid Start can handle this redirect on the server instead of propagating it to the client. 
The data for the redirected page is fetched and streamed to the client in the same http request as the `updateUser` action, rather than the client requiring a separate http request for the redirected page.
```
